Source/Core/DiscIO/FileSystemGCWii.cpp

```diff
@@ -19,83 +21,309 @@
 
 namespace DiscIO
 {
+// Set everything manually
+CFileInfoGCWii::CFileInfoGCWii(const u8* fst, bool wii, u32 fst_size, u32 index, u32 total_file_infos)
+	: m_fst(fst),
+	m_wii(wii),
+	m_fst_size(fst_size),
+	m_index(index),
+	m_total_file_infos(total_file_infos)
+{ }
+
+// For the root object only
+CFileInfoGCWii::CFileInfoGCWii(const u8* fst, bool wii, u32 fst_size)
+	: m_fst(fst),
+	m_wii(wii),
+	m_fst_size(fst_size),
+	m_index(0),
+	m_total_file_infos(GetSize())
+{ }
+
+// Copies data that is common to the whole file system
+CFileInfoGCWii::CFileInfoGCWii(const CFileInfoGCWii& file_info, u32 index)
+	: m_fst(file_info.m_fst),
+	m_wii(file_info.m_wii),
+	m_fst_size(file_info.m_fst_size),
+	m_index(FixIndex(index)),
+	m_total_file_infos(file_info.m_total_file_infos)
+{ }
+
+CFileInfoGCWii::~CFileInfoGCWii()
+{
+}
+
+
+u64 CFileInfoGCWii::GetFSTSize() const
+{
+	return (u64)m_fst_size << (m_wii ? 2 : 0);
```

Comment:
Calling CFileSystemGCWii's GetOffsetShift would require a reference to that object.